src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -611,28 +611,36 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
         {
             // If the job method is an instance method and the job class implements
             // the filter interface, this filter runs first before all other filters
-            TFilter instanceFilter = instance as TFilter;
-            if (instanceFilter != null)
+            if (instance is TFilter instanceFilter)
             {
                 yield return instanceFilter;
             }
 
             // Add any global filters
-            foreach (var filter in globalFunctionFilters.OfType<TFilter>())
+            foreach (var filter in globalFunctionFilters)
```

Comment:
What is the gain here? We're still iterating once through the filters as we were before, right?

Reply:
Missed this! Yes we are iterating but not double iterating - we're not allocating the intermediate OfTypeIterator through LINQ per call ([which has it's own `foreach`](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Linq/src/System/Linq/Cast.cs#L23-L29)). Here's a sample comparison benchmark of the 2 approaches:
```cs
[ShortRunJob]
[MemoryDiagnoser]
public class IteratorBenchmarks
{
	private static List<object> things = new() {
		"foo",
		"bar",
		new object(),
		new AggregateException(),
		new object(),
		"foo",
	};
	
	[Benchmark]
	public void Direct()
	{
		foreach (var obj in things)
		{
			if (obj is string stringObj)
			{
				var i = stringObj;
			}
		}
	}

	[Benchmark]
	public void OfType()
	{
		foreach (var stringObj in things.OfType<string>())
		{
			var i = stringObj;
		}
	}
}
```
And the comparison:
| Method |       Mean |    Error |    StdDev |  Gen 0 | Allocated |
|------- |-----------:|---------:|----------:|-------:|----------:|
| Direct |   6.922 ns | 1.326 ns | 0.0727 ns |      - |         - |
| OfType | 105.029 ns | 4.618 ns | 0.2531 ns | 0.0057 |      96 B |